experimental/oak_async/src/grpc.rs

```diff
@@ -129,3 +145,116 @@ impl<T> Drop for MultiWriter<T> {
         }
     }
 }
+
+/// A streaming gRPC client request.
+///
+/// `T` represents the message type sent in the requests.
+pub struct GrpcRequestStream<T: Decodable> {
+    inner: Peekable<ChannelReadStream<GrpcRequest>>,
+    is_drained: bool,
+    _msg: core::marker::PhantomData<T>,
+}
+
+// Unpin is needed to be able to peek at the first request before passing it
+// on to the handler.
+impl<T: Decodable> Unpin for GrpcRequestStream<T> {}
+
+// This should only be used by the Oak service generator. Not a stable API.
+#[doc(hidden)]
+impl GrpcRequestStream<()> {
+    pub fn new(inner: ChannelReadStream<GrpcRequest>) -> GrpcRequestStream<()> {
```

Comment:
Can this be non-public?

Reply:
Aah yes, this can actually by private now. Thanks for spotting!